requirements/build.txt

```diff
@@ -249,3 +249,14 @@ zope-interface==5.2.0
     # via
     #   -r requirements/requirements.txt
     #   gevent
+
+# The following packages are considered to be unsafe in a requirements file:
```

Comment:
For future reference for other devs, I asked @seanh about the unsafe issue here: https://hypothes-is.slack.com/archives/C1MA4E9B9/p1697538183602879?thread_ts=1696605845.390609&cid=C1MA4E9B9

> But it has a --allow-unsafe option that is currently disabled by default but scheduled to be enabled by default in the next release, and for now they recommend passing that option (and make requirements does pass it). I suspect the formatting differences are to do with that: https://github.com/jazzband/pip-tools?tab=readme-ov-file#deprecations If you read the GitHub issue that links to, pip-compile's own maintainers don't seem to know why those packages were ever considered unsafe, it seems that no one remembers